src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.cs

```diff
@@ -1117,6 +1117,108 @@ public AssemblyReference GetAssemblyReference(AssemblyReferenceHandle handle)
             return new AssemblyReference(this, handle.Value);
         }
 
+        public AssemblyName GetAssemblyName()
+        {
+            StringHandle nameHandle = AssemblyTable.GetName();
+
+            string name = (!nameHandle.IsNil) ? GetString(nameHandle) : null;
+
+            Version version = AssemblyTable.GetVersion();
+
+            StringHandle cultureHandle = AssemblyTable.GetCulture();
+            string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : null;
+
+            BlobHandle publicKeyHandle = AssemblyTable.GetPublicKey();
+            byte[] publicKeyOrToken = !publicKeyHandle.IsNil ? GetBlobBytes(publicKeyHandle) : null;
+
+            var assemblyName = new AssemblyName(name)
+            {
+                Version = version,
+                CultureName = cultureName
+            };
+
+            assemblyName.SetPublicKey(publicKeyOrToken);
+            SetAssemblyNameFlags(assemblyName);
+            SetAssemblyNameHashAlgorithm(assemblyName);
+
+            return assemblyName;
+        }
+
+        private void SetAssemblyNameFlags(AssemblyName assemblyName)
+        {
+            var flagMappings = new Dictionary<AssemblyFlags, AssemblyNameFlags>()
+            {
+                [AssemblyFlags.PublicKey] = AssemblyNameFlags.PublicKey,
+                [AssemblyFlags.Retargetable] = AssemblyNameFlags.Retargetable,
+                [AssemblyFlags.EnableJitCompileTracking] = AssemblyNameFlags.EnableJITcompileTracking
+            };
+
+            AssemblyFlags flags = AssemblyTable.GetFlags();
+
+            foreach (var flagItem in flagMappings)
+            {
+                bool flagExists = (flags & flagItem.Key) != 0;
+
+                if (flagExists)
+                {
+                    assemblyName.Flags |= flagItem.Value;
+                }
+                else
+                {
+                    assemblyName.Flags &= ~flagItem.Value;
+                }
+            }
+            
+            SetJitCompileOptimizerEnabled(assemblyName, flags);
+
+            // the following two flags have no match in AssemblyNameFlags
+            bool hasContentTypeMask = (flags & AssemblyFlags.ContentTypeMask) != 0;
+            bool hasWindowsRuntime = (flags & AssemblyFlags.WindowsRuntime) != 0;
+        }
+
+        private void SetAssemblyNameHashAlgorithm(AssemblyName assemblyName)
```

Comment:
same comment here.

optionally perhaps I could move this to a AssemblyNameUtils helper class instead